apps/condo/domains/acquiring/schema/MultiPayment.js

```diff
@@ -32,33 +56,48 @@ const MultiPayment = new GQLListSchema('MultiPayment', {
             type: Virtual,
             resolver: (item) => {
                 const explicitFee = get(item, 'explicitFee')
-                const floatFee = parseFloat(explicitFee || '0')
-                const floatAmount = parseFloat(item.amountWithOutExplicitFee)
-                return String(floatAmount + floatFee)
+                const floatFee = new Big(explicitFee || '0')
+                const floatAmount = new Big(item.amountWithoutExplicitFee)
+                return floatAmount.plus(floatFee).toString()
             },
         },
 
         explicitFee: {
-            ...MONEY_AMOUNT_FIELD,
+            ...NON_NEGATIVE_MONEY_FIELD,
             schemaDoc: 'The amount of commission which resident pay on top of amount',
             isRequired: false,
         },
 
         serviceFee: {
-            ...MONEY_AMOUNT_FIELD,
+            ...NON_NEGATIVE_MONEY_FIELD,
             schemaDoc: 'The amount of money charged by service (Doma) for the provision of service. Can be explicit or implicit',
             isRequired: false,
         },
 
         implicitFee: {
-            ...MONEY_AMOUNT_FIELD,
+            ...NON_NEGATIVE_MONEY_FIELD,
             schemaDoc: 'Total amount of money charged from recipients of multipayment as fee for transaction',
             isRequired: false,
             access: { read: access.canReadMultiPaymentsSensitiveData },
+            hooks: {
+                validateInput: ({ resolvedData, addFieldValidationError, fieldPath, listKey, operation, existingItem }) => {
+                    if (resolvedData.hasOwnProperty(fieldPath) && resolvedData[fieldPath] !== null) {
```

Comment:
`if (!isNull(get(resolvedData, fieldPath))) {`

also please make sure you're handling case for `udefined`, because it's a valid value for update operation, and means nothing changed

Reply:
You're not correct here....Let me explain how keystone resolvers work:
1. If field has no changes => it will not get in`resolvedData`, so `resolvedData` will not have property `fieldPath`
2. If we're trying to delete value, `resolvedData[fieldPath]` will be null
3. If `resolvedData` contains non-null `fieldPath` => it has correct type and already passed type checks

So, my checks are explicitly saying something like "Process only if field is changing and value is not going to be deleted"